modules/user_accounts/php/NDB_Form_user_accounts.class.inc

```diff
@@ -829,6 +835,71 @@ class NDB_Form_User_Accounts extends NDB_Form
     }
 
     /**
+    * Query database on UserID to check that password has doesn't
+    * exist and account is still pending approval, to ensure the
+    * users with account activity can't be rejected by setting
+    * their status to pending.
+    *
+    * @param string $id identifier of account to verify
+    *
+    * @return boolean true is Password_hash is null and
+    * Pending_approval is Yes, false otherwsie
+    */
+    function _canRejectAccount($id)
+    {
+        $DB     = Database::singleton();
+        $fields =  $DB->pselect(
+            "SELECT Password_hash, Pending_approval
+                   FROM users
+                   WHERE UserID=:id",
+            array("id" => $id)
+        );
+        if (is_null($fields[0]['Password_hash'])
+            && $fields[0]['Pending_approval']=="Y"
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+    * Send DB query to remove user based on userID through a GET
+    * request.
+    * The reject user button populates on edit_users page, therefore
+    * any user that has access to the button has user_accounts
+    * privileges.
+    * The URL is redirected to user_account after the operation
+    *
+    * @return void
+    */
+    // @codingStandardsIgnoreStart
+    function reject_user()
+    {
+    // @codingStandardsIgnoreEnd
+        $DB     = Database::singleton();
+        $editor = User::singleton();
+
+        $redirect = $baseURL . "/user_accounts";
+
+        // The $editor variable is set by the edit_user script.
+        // Checking that this is set ensures that the user is
+        // accessing this function through the authorized button
+        // and not by pointing their URL directly to this script.
+        if (isset($editor)) {
+            if ($this->identifier!='') {
```

Comment:
This deletion logic belongs in the `_process` function, not the `reject_user` function (which is for the view logic of the reject_user page).

Deleting from the users table is also very dangerous, this should set the Active flag to 'N', so there's no chance of irrecoverable data loss caused accidentally deleting the wrong user (and any data that may have a foreign key pointing at it.